### ISSUE TITLE
Documentation Not Helpful

### ISSUE THREAD

**Issue by mslinn** [rails/thor#848]
The official documentation for Thor at http://whatisthor.com/ consists of one short web page, and is woefully incomplete. For example, it makes no mention of templates. This tiny website provides very little value.

The wiki at https://github.com/rails/thor/wiki/Method-Options says:

> NOTE: a good part of the documentation here is outdated, see the website instead: http://whatisthor.com/

However, http://whatisthor.com/ also makes no mention of `method_options` either.

Thor is well-designed, well-written, and well-maintained. Its documentation, however, presents a significant barrier to those who are keen to learn it (like me). 

Save the expense of the yearly registration fee for http://whatisthor.com/, and please maintain the documentation on GitHub instead.

**Comment by rafaelfranca** [comment]
Agree. Mind to take the opportunity that you are writing about your experiences and move the documentation in the website to the README of the project and remove all mentions to http://whatisthor.com/?

**Comment by mattbrictson** [comment]
> If someone provides me with a sentence or two about the high-level purpose and motivation of the `thor` command

Does this sound accurate?

Thor is a gem for building CLIs in Ruby. It serves multiple use-cases:

1. The Thor gem adds a `thor` command to your shell. Much like Rake, you can define command-line scripts in Ruby, place them in a `Thorfile`, and then execute them using `thor`. Compared to Rake, Thor offers a richer, modern set of CLI features and a more intuitive programming model.
2. Thor also provides tools for building entire packaged CLIs. For example, the CLIs of [`mrsk`](https://github.com/mrsked/mrsk), [`bundler-audit`](https://github.com/rubysec/bundler-audit), and [`good_job`](https://github.com/bensheldon/good_job#command-line-options) are all built using Thor.
3. Thor contains a powerful library of methods for generating files and directory trees from Ruby and ERB templates, as well as conveniences for prompting for user input and printing output on the console. These methods can be used within Thor CLIs or from your own Ruby code.
4. Finally, Thor is the foundation of the [Generator and App Template](https://guides.rubyonrails.org/generators.html) system in Rails. Commands like `rails new` and `rails generate devise:install` use Thor's library of template methods behind the scenes.

**Comment by mattbrictson** [comment]
Maybe a more concise way of saying it is:

> Thor is a task-runner, a framework for building CLIs, and a library of template and generator methods.

**Comment by mslinn** [comment]
Proper documentation would reorder those, such that the briefest summary was presented first, then a longer exposition, then as much additional detail as was appropriate.
